lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/table/TableRowTruthDAOImpl.java

```diff
@@ -67,7 +67,7 @@
  * @author John
  * 
  */
-public class TableRowTruthDAOImpl implements TableRowTruthDAO {
+public abstract class TableRowTruthDAOImpl implements TableRowTruthDAO {
```

Comment:
The CachingTableRowTruthDAOImpl  uses the cache to optimize the check.  The original version of the method in TableRowTruthDAOImpl would scan through all changes in S3 with a version number greater than the version number of the update.